### PR TITLE
Add HOMEBREW_DISABLE_LOAD_FORMULA_MESSAGE

### PR DESCRIPTION
**Pull request by maxim-belkin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

I suggest to add a new env. variable to turn off those `brew.rb loading <formula-name>` messages. This should be useful to Homebrew developers working on issues unrelated to the loading step, e.g. `brew livecheck --debug --tap homebrew/core`.

### CODE REVIEW

**Comment by maxim-belkin** [comment]
> I don't think we should be playing whack-a-mole with selectively disabling messages only displayed in verbose or debug mode.

I agree with that in general (disabling specific messages using an env. var does look weird), but scrolling through the "loading" messages  when dealing with anything that loads entire taps gets pretty annoying very quickly... especially when you know that this "loading" step is not what you're looking for. Universal debug mode is better than no debug mode but... it makes debugging job a tiny bit less enjoyable than it usually is.

So, debugging modes (at least in some applications) have several levels and messages are printed (or not) depending on that level. The challenge with Homebrew is that it isn't a one-big-app that does one-big-thing. It has many components that load and use one another and it brings up the question of whether more fine-grained debugging regime can be devised. I can come up with a number of (crazy) ideas, e.g.:
* new `--debug-level=` parameger
* calling out which named debugging information to print, e.g. `--debug-components=something,all`
* calling out which Homebrew files/directories to debug, e.g. `--debug-files=formula_installer.rb,`

but these might not be the best and someone else could probably come up with a better idea.
So, shall I close this one?

**Comment by MikeMcQuaid** [comment]
> So, shall I close this one?

I'd vote 👍 but let's leave it open for now and see what other @Homebrew/maintainers think and if they need/want something like this. Thanks @maxim-belkin!

File: Library/Homebrew/env_config.rb [hunk 1]
```diff
@@ -5,166 +5,171 @@ module EnvConfig
...
-      HOMEBREW_GIT_NAME:                  {
+      HOMEBREW_GIT_NAME:                     {
         description: "Set the Git author and committer email to this value.",
       },
-      HOMEBREW_INSTALL_BADGE:             {
+      HOMEBREW_INSTALL_BADGE:                {
         description:  "Print this text before the installation summary of each successful build.",
         default_text: 'The "Beer Mug" emoji.',
-        default:      "🍺",
+        default:      "Ã°ÂÂÂº",
```
Comment (bayandin) [inline Library/Homebrew/env_config.rb:164]: Something went wrong

Comment (maxim-belkin) [reply Library/Homebrew/env_config.rb:164]: ah, that's `brew style --fix`. I didn't touch here anything besides lines 94-98

**Comment by claui** [comment]
Depending on how often you run this:
would appending `2>&1 | grep -ivE 'loading|^$'` help?

**Comment by maxim-belkin** [comment]
> Depending on how often you run this:
> would appending `2>&1 | grep -ivE 'loading|^$'` help?

Kind of. We'd have to make sure we don't block/ignore other messages that have 'Loading' in them. And btw, I don't think empty lines are bad :)

**Comment by claui** [comment]
> > Depending on how often you run this:
> > would appending `2>&1 | grep -ivE 'loading|^$'` help?
> 
> Kind of. We'd have to make sure we don't block/ignore other messages that have 'Loading' in them. And btw, I don't think empty lines are bad :)

Fair enough.

**Comment by MikeMcQuaid** [comment]
> Kind of. We'd have to make sure we don't block/ignore other messages that have 'Loading' in them. And btw, I don't think empty lines are bad :)

I'd be open to changing this message to be more easily greppable e.g. `Loading formula:` or something.

**Comment by maxim-belkin** [comment]
> I'd be open to changing this message to be more easily greppable e.g. `Loading formula:` or something.

Well, even now it can be grepped with `(Formulary::FormulaLoader): loading`. But I think a programmatic approach would be better. Otherwise, why does `brew home` exist if we can `open $(brew info gcc | grep "^[fht]\+tp")` ?

**Comment by vitorgalvao** [comment]
> Otherwise, why does `brew home` exist if we can `open $(brew info gcc | grep "^[fht]\+tp")` ?

`brew home` is a user-facing command that needs to be simple and memorable, so even the less technically inclined can use it without confusion. Its simplicity affects everyone.

This proposal only affects a small subset of developers some of the time. If this can be easily achievable with a single pipe to `grep` (which can be further abstracted in the shell), I also think it’s reasonable to do it that way rather than maintaining a new environment variable with such specific behaviour.

**Comment by maxim-belkin** [comment]
I see there is a consensus that these debugging messages are useful to you all. I'll close this PR but I do hope that someone will continue to think about improving debugging system within Homebrew. Even `--debug=basic,intermediate,all` would be a great thing to have. 

Thank y'all for the responses. ❤️

**Pull request closed without merging**